examples/tests/kv_interactive/main.c

```diff
@@ -0,0 +1,144 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <console.h>
+#include <kv_system.h>
+
+#define KEY_LEN  64
+#define DATA_LEN 64
+
+uint8_t key_buf[KEY_LEN];
+uint8_t data_buf[DATA_LEN];
+uint8_t value_buf[DATA_LEN];
+
+char read_buf[64];
+
+static int get_command(void) {
+  int idx = 0;
+  memset(read_buf, 0, 64);
+
+  // Read in characters until a new line
+  while (1) {
+    int c = getch();
+
+    if (c == RETURNCODE_FAIL) {
+      printf("\ngetch() failed!\n");
+      return 0;
+
+    } else {
+      char in = c;
+
+      if (in == '\n' || in == '\r') {
+        return 0;
+
+      } else {
+        // If this is a valid number record it
+        read_buf[idx] = in;
+        idx += 1;
+
+        // If our buffer is full, quit
+        if (idx >= 64) {
+          return 0;
+        }
+      }
+    }
+  }
+}
+
+static int find_end(int start) {
+  for (int i = start; i < DATA_LEN; i++) {
```

Comment:
This is a different length then the actual `read_buf`. A user could reduce the size of `read_buf` but not `DATA_LEN` and this won't work. There should be more length and access checks to avoid overflows

Reply:
> char read_buf[DATA_LEN];

Looks like this is now sized to match.